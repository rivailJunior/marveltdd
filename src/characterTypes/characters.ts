export type CharacterDataContainer = {
    offset: number
    limit: number
    total: number
    count: number
    results: [Character]
}

export type Character = {
    id: number
    name: string
    description: string
    modified: Date
    resourceURI: String
    urls?: string[] | null
    thumbnail?: Image
    comics?: ComicList[] | any
    stories?: StoryList[] | any
    events?: EventList[] | any
    series?: SeriesList[] | any
}

export type Image = {
    path: string
    extension: string
}

export type ComicList = {
    available: number
    returned: number
    collectionURI: string
    items: [ComicSummary]
}
export type ComicSummary = {
    resourceURI: string
    name: string
}

export type StoryList = {
    available: number
    returned: number
    collectionURI: string
    items: [StorySummary]
}

export type StorySummary = {
    resourceURI: string
    name: string
    type: string
}

export type EventList = {
    items?: EventSummary[] | null
    available: number
    returned: number
    collectionURI: string
}

export type EventSummary = {
    resourceURI: string
    name: string
}

export type SeriesList = {
    items?: SeriesSummary[] | null
    available: number
    returned: number
    collectionURI: string
}

export type SeriesSummary = {
    resourceURI: string
    name: string
}