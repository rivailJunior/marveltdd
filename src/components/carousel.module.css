.showCarousel {
    display: block;
}

.hideCarousel {
    display: none;
}