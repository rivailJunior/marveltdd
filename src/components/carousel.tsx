import { useState } from "react";
import { Carousel } from "react-bootstrap";
import styles from "./carousel.module.css";

type iCarouselProps = {
    data: [any];
    type: string;
};
export function CarouselInfor({ data, type }: iCarouselProps): JSX.Element {
    const [index, setIndex] = useState(0);

    const handleSelect = (selectedIndex: number, e: any) => {
        setIndex(selectedIndex);
    };

    const showErr = !data.length ? styles.showCarousel : styles.hideCarousel;
    const showCarousel = data.length
        ? styles.showCarousel
        : styles.hideCarousel;
    return (
        <>
            <div className={showErr}>Ops. Nenhuma {type} encontrado</div>
            <Carousel
                className={showCarousel}
                activeIndex={index}
                onSelect={handleSelect}
            >
                {data.map((item, index) => {
                    <Carousel.Item key={index}>
                        <img
                            className="d-block w-100"
                            src="holder.js/800x400?text=First slide&bg=373940"
                            alt="First slide"
                        />
                        <Carousel.Caption>
                            <h3>First slide label</h3>
                            <p>
                                Nulla vitae elit libero, a pharetra augue mollis
                                interdum.
                            </p>
                        </Carousel.Caption>
                    </Carousel.Item>;
                })}
            </Carousel>
        </>
    );
}
