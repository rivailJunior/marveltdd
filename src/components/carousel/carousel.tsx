import { useState } from "react";
import { Carousel } from "react-bootstrap";

type iCarouselProps = {
    data: [] | any;
    type: string;
};
export function CarouselInfor({ data, type }: iCarouselProps): JSX.Element {
    const [index, setIndex] = useState(0);

    const handleSelect = (selectedIndex: number, e: any) => {
        setIndex(selectedIndex);
    };

    return (
        <>
            {!data.length ? (
                <div>Ops. Nenhuma {type} encontrado</div>
            ) : (
                <>
                    <div
                        className="labelHeader"
                        style={{ fontSize: 20, fontWeight: 500 }}
                    >
                        {type}
                    </div>
                    <Carousel activeIndex={index} onSelect={handleSelect}>
                        {data.map((item: any, id: number) => {
                            return (
                                <Carousel.Item
                                    key={id}
                                    data-testid="carouselItem"
                                >
                                    <img
                                        className="d-block w-100"
                                        src={`${ item?.thumbnail?.path }.${ item?.thumbnail?.extension }`}
                                        alt="First slide"
                                    />
                                    <Carousel.Caption>
                                        <h3>{item?.title}</h3>
                                        <p>{item?.endYear}</p>
                                    </Carousel.Caption>
                                </Carousel.Item>
                            );
                        })}
                    </Carousel>
                </>
            )}
        </>
    );
}
