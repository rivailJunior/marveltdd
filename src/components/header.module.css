.marvelHeader {
    /* padding: 10%; */
    top: 0px;
    left: 0px;
    width: 100%;
    height: 64px;
    background: #FFFFFF 0% 0% no-repeat padding-box;
    opacity: 1;
}

.labelName {
    text-align: left;
    font: normal normal bold 14px/20px PT Sans Caption;
    letter-spacing: 0px;
    color: #555555;
}

.informationProps {
    height: 100%;
    width: 100%;
    background-color: lightcoral;
}