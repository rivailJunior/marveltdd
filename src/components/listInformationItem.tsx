import styles from "./marvelList.module.css";

type iListProps = {
    list: [any] | any;
    tagName: string;
    information?: string;
};
export const ListInformationItem = ({
    list,
    tagName,
    information,
}: iListProps): JSX.Element => {
    return list?.length > 0 ? (
        <ul className={styles.list}>
            {list.slice(0, 3).map((item: any, index: number) => {
                return (
                    <li data-testid="informationLi" key={index}>
                        {item[tagName]}
                    </li>
                );
            })}
        </ul>
    ) : (
        <div className={styles.defaultLabel}>
            Ops. {information} n√£o encontrados!
        </div>
    );
};
