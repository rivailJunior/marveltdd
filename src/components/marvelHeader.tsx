import { Navbar } from "react-bootstrap";
import styles from "./header.module.css";

type iHeaderProps = {
    leftLabel?: string;
    userName?: string;
    descriptionRight?: string;
};

const MarvelHeader = ({
    leftLabel,
    userName,
    descriptionRight,
}: iHeaderProps): JSX.Element => {
    const NoPropsDivInfor = () => {
        return (
            <div
                className={styles.informationProps}
                data-testid="informationProps"
            >
                ops!
            </div>
        );
    };

    return (
        <Navbar
            expand="lg"
            variant="light"
            bg="light"
            className={styles.marvelHeader}
        >
            <Navbar.Brand href="#home">
                {!leftLabel?.length ? NoPropsDivInfor() : leftLabel}
            </Navbar.Brand>
            <Navbar.Toggle />

            <Navbar.Collapse className="justify-content-end">
                <Navbar.Text className={styles.labelName}>
                    {!userName?.length ? NoPropsDivInfor() : userName}
                    {!descriptionRight?.length ? (
                        NoPropsDivInfor()
                    ) : (
                        <a href="#login">{descriptionRight}</a>
                    )}
                </Navbar.Text>
            </Navbar.Collapse>
        </Navbar>
    );
};

export { MarvelHeader };
