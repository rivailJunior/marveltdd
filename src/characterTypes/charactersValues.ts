
import { Character, CharacterDataContainer, ComicSummary, ComicList, StorySummary, StoryList, EventSummary, EventList, SeriesSummary, SeriesList, Image } from './characters';

export const ComicSummaryValue: ComicSummary = {
    resourceURI: "",
    name: ""
}

export const ImageValue: Image = {
    path: '',
    extension: ''
}

export const ComicListValue: ComicList = {
    available: 0,
    returned: 0,
    collectionURI: "",
    items: [ComicSummaryValue]
}

export const StorySummaryValue: StorySummary = {
    resourceURI: "",
    name: "",
    type: "",
}

export const StoryListValue: StoryList = {
    available: 0,
    returned: 0,
    collectionURI: "",
    items: [StorySummaryValue]
}

export const EventSummaryValue: EventSummary = {
    resourceURI: "",
    name: "",
}

export const EventListValue: EventList = {
    items: [EventSummaryValue],
    available: 0,
    returned: 0,
    collectionURI: "",
}

export const SeriesSummaryValue: SeriesSummary = {
    resourceURI: "",
    name: ""
}

export const SeriesListValue: SeriesList = {
    items: [SeriesSummaryValue],
    available: 0,
    returned: 0,
    collectionURI: ""

}

export const CharacterValue: Character = {
    id: 0,
    name: '',
    description: '',
    modified: new Date(),
    resourceURI: '',
    urls: [''],
    thumbnail: ImageValue,
    comics: [ComicListValue],
    stories: [StoryListValue],
    events: [EventListValue],
    series: [SeriesListValue]
}


export const CharacterDataContainerValue: CharacterDataContainer = {
    offset: 0,
    limit: 0,
    total: 0,
    count: 0,
    results: [CharacterValue]
}

