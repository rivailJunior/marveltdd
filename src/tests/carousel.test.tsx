import React from "react";
import {
    render,
    cleanup,
    fireEvent,
    getByTestId,
    getAllByTestId,
} from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { MemoryRouter } from "react-router-dom";
import { CarouselInfor } from "../components/carousel";
import { responseSeriesMock } from "../characterTypes/requestValueMock";

afterEach(cleanup);

describe("`<CarouselInfor>`", () => {
    test("Render correctly", () => {
        const container = renderer.create(
            <CarouselInfor data={[1, 2, 3]} type="Series" />
        );
        container.toJSON();
        expect(container).toMatchSnapshot();
    });

    test("Render with no item to be shown ", () => {
        const { getByText } = render(<CarouselInfor data={[]} type="Series" />);
        expect(
            getByText(/Ops. Nenhuma Series encontrado/i)
        ).toBeInTheDocument();
    });

    test("Render Series ", () => {
        const { getByText, getAllByTestId } = render(
            <CarouselInfor
                data={responseSeriesMock.data.results}
                type="Series"
            />
        );
        expect(getByText(/series/i)).toBeInTheDocument();
        expect(getAllByTestId("carouselItem")).toHaveLength(
            responseSeriesMock.data.results.length
        );
    });
});
