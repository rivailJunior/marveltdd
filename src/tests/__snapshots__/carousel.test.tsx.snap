// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`<CarouselInfor>\` Render correctly 1`] = `
Array [
  <div
    className="labelHeader"
    style={
      Object {
        "fontSize": 20,
        "fontWeight": 500,
      }
    }
  >
    Series
  </div>,
  <div
    className="carousel slide"
    onKeyDown={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <ol
      className="carousel-indicators"
    >
      <li
        className="active"
        onClick={[Function]}
      />
      <li
        onClick={[Function]}
      />
      <li
        onClick={[Function]}
      />
    </ol>
    <div
      className="carousel-inner"
    >
      <div
        className="active carousel-item"
        data-testid="carouselItem"
      >
        <img
          alt="First slide"
          className="d-block w-100"
          src="undefined.undefined"
        />
        <div
          className="carousel-caption"
        >
          <h3 />
          <p />
        </div>
      </div>
      <div
        className="carousel-item"
        data-testid="carouselItem"
      >
        <img
          alt="First slide"
          className="d-block w-100"
          src="undefined.undefined"
        />
        <div
          className="carousel-caption"
        >
          <h3 />
          <p />
        </div>
      </div>
      <div
        className="carousel-item"
        data-testid="carouselItem"
      >
        <img
          alt="First slide"
          className="d-block w-100"
          src="undefined.undefined"
        />
        <div
          className="carousel-caption"
        >
          <h3 />
          <p />
        </div>
      </div>
    </div>
    <a
      className="carousel-control-prev"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-prev-icon"
      />
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
    <a
      className="carousel-control-next"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-next-icon"
      />
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </div>,
]
`;
